aws-android-sdk-iot/src/test/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManagerTest.java

```diff
@@ -3021,6 +3021,37 @@ public void testEnableMetricsCollection() {
         assertFalse(testClient.isMetricsEnabled());
     }
 
+    @Test
+    public void testMqttSessionPresent() throws MqttException {
+        // Test if sessionPresent flag from AWSIotMqttManager is correctly passed from the MqttClient
+        MockMqttClient mockClient = new MockMqttClient();
+        final boolean actualSessionPresent;
+        actualSessionPresent = mockClient.testToken.getSessionPresent();
+
+        final AWSIotMqttManager testSessionPresentFlag = new AWSIotMqttManager("test-sessionPresent-flag",
+                Region.getRegion(Regions.US_EAST_1), TEST_ENDPOINT_PREFIX);
+        testSessionPresentFlag.setMqttClient(mockClient);
+
+        KeyStore testKeystore = AWSIotKeystoreHelper.getIotKeystore(CERT_ID, KEYSTORE_PATH,
+                KEYSTORE_NAME, KEYSTORE_PASSWORD);
+
+        testSessionPresentFlag.connect(testKeystore, new AWSIotMqttClientStatusCallback() {
+            @Override
+            public void onStatusChanged(AWSIotMqttClientStatus status, Throwable throwable) {
+                if (status == AWSIotMqttClientStatus.Connecting) {
+                    System.out.println("Client persistent-client-id connecton status: " + status);
+                    System.out.println("getSessionPresent: " + testSessionPresentFlag.getSessionPresent());
+                    // Compare sessionPresent flag from AWSIotMqttManager with the actual one
+                    assertEquals(testSessionPresentFlag.getSessionPresent(), actualSessionPresent);
```

Comment:
you don't want to do you assertion inside the callback, because the test might just finish running before the callback is called. 
You should try using a semaphore or simple countdown latch to make sure that the callback is triggered and do a the assertion after `AWSIotMqttManager#connect()`.